### PR TITLE
chore: add deployment script for GroupSync

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Add script to help with scaling GroupSync for private-ip use case.

### CODE REVIEW

File: scripts/build-and-deploy-private-ip.sh [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+######################## IAM and DB ROLE CONFIGURATION ########################
+# grant Cloud SQL Instance User role to all IAM group emails (so users of Group can inherit)
+for GROUP in $IAM_GROUPS;
+do
+  gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member="group:$GROUP" \
+    --role="roles/cloudsql.instanceUser"
+done
+./
```
Comment (enocom) [inline scripts/build-and-deploy-private-ip.sh:91]: This looks like a typo.

File: scripts/build-and-deploy-private-ip.sh [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+######################## Cloud Scheduler variables ########################
+export PATH_TO_JSON="" # file path to JSON payload containing instance-to-group mappings for Cloud Scheduler
+export SCHEDULE="*/10 * * * *" # schedule how often GroupSync Cloud Scheduler is called (defaults to 10 mins)
+
+# load IAM Groups and Cloud SQL Instance names from JSON payload
+IAM_GROUPS=`cat $PATH_TO_JSON | jq '.iam_groups' | tr -d '[],"'`
+SQL_INSTANCES=`cat $PATH_TO_JSON | jq '.sql_instances' | tr -d '[],"'`
+
+# check if required variables are set, otherwise give error and exit
+declare -a vars=(PROJECT_ID REGION PATH_TO_JSON IAM_GROUPS SQL_INSTANCES CONNECTOR_NAME HOST_PROJECT_ID SUBNET)
```
Comment (enocom) [inline scripts/build-and-deploy-private-ip.sh:40]: Would this be simpler as the following?

``` sh
for var_name in A B C D
do
  if [[ -z "${var_name}" ]]; then
    # not defined
  fi
done
```

https://stackoverflow.com/a/39296583/927514

**Review by enocom — APPROVED** [review]
LGTM

A few minor comments.

File: scripts/build-and-deploy-private-ip.sh (hunk 2 above)
Comment (jackwotherspoon) [reply scripts/build-and-deploy-private-ip.sh:40]: This didn't seem to work for me, will leave it as is for now.

**Pull request merged**